### PR TITLE
Replace GET parameters when changed in form

### PR DESCRIPTION
**Pull request by wouterj**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #13962 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/Form.php [hunk 1]
```diff
@@ -197,9 +197,21 @@ public function getUri()
...
         $uri = parent::getUri();
 
-        if (!in_array($this->getMethod(), array('POST', 'PUT', 'DELETE', 'PATCH')) && $queryString = http_build_query($this->getValues(), null, '&')) {
-            $sep = false === strpos($uri, '?') ? '?' : '&';
-            $uri .= $sep.$queryString;
+        if (!in_array($this->getMethod(), array('POST', 'PUT', 'DELETE', 'PATCH'))) {
+            $query = parse_url($uri, PHP_URL_QUERY);
+            $currentParameters = array();
+            if ($query) {
+                foreach (explode('&', $query) as $param) {
```
Comment (javiereguiluz) [inline src/Symfony/Component/DomCrawler/Form.php]: Could it be possible to replace these two combined `explode` functions + `foreach` by a call to the PHP [parse_str() function](http://php.net/parse-str)?

Comment (WouterJ) [inline src/Symfony/Component/DomCrawler/Form.php]: Unfortunately, as soon on PHP.net:

 > Parses str as if it were the query string passed via a URL and **sets variables in the current scope.**

There is no easy (nor quick) way to find out which variables were added in the current scope IRCC

Comment (javiereguiluz) [inline src/Symfony/Component/DomCrawler/Form.php]: But if you use the second optional argument of the function, the variables are stored in an array, right?

Comment (WouterJ) [inline src/Symfony/Component/DomCrawler/Form.php]: Indeed, I've updated the PR.

**Comment by fabpot** [comment]
Thank you @WouterJ.

**Pull request merged**